File: replacer-backend/src/main/java/es/bvalero/replacer/page/review/ReviewReplacement.java
package es.bvalero.replacer.page.review;

import io.swagger.v3.oas.annotations.media.Schema;
import java.util.Collection;
import lombok.Value;
import org.springframework.lang.NonNull;

@Schema(description = "Replacement to review")
@Value(staticConstructor = "of")
class ReviewReplacement {

    @Schema(description = "Position of the replacement in the content", required = true, example = "1776")
    int start;

    @Schema(description = "Text of the replacement", example = "aún", required = true)
    @NonNull
    String text;

    @Schema(description = "Kind of the replacement", example = "2", required = true)
    byte kind;

    @Schema(description = "Subtype of the replacement", example = "habia", required = true)
    @NonNull
    String subtype;

    @Schema(description = "Collection of suggestions to fix the replacement", required = true)
    @NonNull
    Collection<ReviewSuggestion> suggestions;
}

File: replacer-frontend/src/app/api/models/review-replacement.ts
/* tslint:disable */
/* eslint-disable */
import { ReviewSuggestion } from './review-suggestion';

/**
 * Replacement to review
 */
export interface ReviewReplacement {

  /**
   * Kind of the replacement
   */
  kind: number;

  /**
   * Position of the replacement in the content
   */
  start: number;

  /**
   * Subtype of the replacement
   */
  subtype: string;

  /**
   * Collection of suggestions to fix the replacement
   */
  suggestions: Array<ReviewSuggestion>;

  /**
   * Text of the replacement
   */
  text: string;
}
